server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/references/ElementOccurrences.java

```diff
@@ -45,10 +45,20 @@ public ElementOccurrences(SourceUnitGraph sourceUnitGraph) {
   public @NonNull List<Location> findDefinitions(
       @NonNull CobolDocumentModel document, @NonNull TextDocumentPositionParams position) {
     String uri = position.getTextDocument().getUri();
-    return SymbolsRepository.findElementByPosition(
-            uri, document.getLastAnalysisResult(), position.getPosition())
-        .map(DefinedAndUsedStructure::getDefinitions)
-        .orElse(Collections.emptyList());
+    return getEmbeddedCopybookLocation(position, uri)
+        .orElse(
+            SymbolsRepository.findElementByPosition(
+                    uri, document.getLastAnalysisResult(), position.getPosition())
+                .map(DefinedAndUsedStructure::getDefinitions)
+                .orElse(Collections.emptyList()));
```

Comment:
Should this be the other way around? `SymbolsRepository.findElementByPosition(...).orElse(getEmbeddedCopybookLocation(position, uri))`?

Reply:
adjusting the copyNode to right location in the transformTree#addCopyNodes helps. Previously, nested copyNodes was added as a child to rootNode